resources/resource_utils.py

```diff
@@ -0,0 +1,101 @@
+from typing import Optional, Dict, Any
+import hail as hl
+
+
+# Resource classes
+class BaseResource:
+    """
+    Generic abstract resource class.
+    """
+
+
+    def __init__(self, path: str, source_path: Optional[str] = None, attributes: Dict[str, Any] = None):
+        """
+        Creates a Resource
+
+        :param str path: The resource path
+        :param str source_path: The path for the resource source when it was originally imported
+        :param dict of str attributes: Additional attributes for the resource
+        """
+        if type(self) is BaseResource:
+            raise TypeError("Can't instantiate abstract class BaseResource")
+
+        self.source_path = source_path
+        self.path = path
+        self.attributes = attributes
+
+    def __repr__(self):
+        attr_str = [fr'path={self.path}']
```

Comment:
For my own knowledge why use the raw string?

Reply:
sorry -- this is a remnant of intermediate stuff I was doing, thanks for catching this.